src/components/Popups/PopupCursor.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import { FormattedMessage } from 'react-intl';
+import messages from './messages';
+import PopupBase from './PopupBase';
+import { Options } from './Popper';
+import './PopupCursor.scss';
+
+export type Props = {
+    isActive: boolean;
+};
+
+const options: Partial<Options> = {
+    modifiers: [
+        {
+            name: 'offset',
+            options: {
+                offset: [0, 20],
+            },
+        },
+    ],
+    placement: 'right',
+};
+
+const generateGetBoundingClientRect = (x = 0, y = 0) => () => ({
+    width: 0,
+    height: 0,
+    top: y,
+    right: x,
+    bottom: y,
+    left: x,
+});
+
+export default class PopupCursor extends React.PureComponent<Props> {
+    popupRef = React.createRef<PopupBase>();
+
+    isDuplicated = false;
+
+    virtualElement = {
+        getBoundingClientRect: generateGetBoundingClientRect(),
+    };
+
+    constructor(props: Props) {
+        super(props);
+        if (document.querySelector('.ba-PopupCursor')) {
```

Comment:
We should avoid using query selectors in React components. We should be able to accomplish something like this via refs.